blueprints/data-solutions/data-platform-foundations/demo/dataflow-csv2bq/README.md

```diff
@@ -0,0 +1,58 @@
+## Pipeline summary
+This demo serves as a simple example of building and launching a Flex Template Dataflow pipeline. The code mainly focuses on reading a CSV file as input along with a JSON schema file as side input. The pipeline Parses both inputs and writes the data to the relevant BigQuery table while applying the schema passed from input.
+
+![Dataflow pipeline overview](../../images/df_demo_pipeline.png "Dataflow pipeline overview")
+
+## Example build run
+
+Below is an example for triggering the Dataflow flex template build pipeline defined in `cloudbuild.yaml`. The Terraform output provides an example as well filled with the parameters values based on the generated resources in the data platform.
+
+```
+GCP_PROJECT="ff-orc"
+TEMPLATE_IMAGE="[REGION].pkg.dev/[ORCHESTRATION-PROJECT]/[REPOSITORY]/csv2bq:latest"
+TEMPLATE_PATH="gs://[DATAFLOW-TEMPLATE-BUCKEt]/csv2bq.json"
+STAGIN_PATH="gs://[ORCHESTRATION-STAGING-BUCKET]/build"
+LOG_PATH="gs://[ORCHESTRATION-LOGS-BUCKET]/logs"
+REGION="[YOUR-REGION]"
+
+gcloud builds submit \
+           --config=cloudbuild.yaml \
+           --project=$GCP_PROJECT \
+           --region="europe-west1" \
+           --gcs-log-dir=$LOG_PATH \
+           --gcs-source-staging-dir=$STAGIN_PATH \
+           --substitutions=_TEMPLATE_IMAGE=$TEMPLATE_IMAGE,_TEMPLATE_PATH=$TEMPLATE_PATH,_DOCKER_DIR="."
+```
+
+## Example Dataflow pipeline launch in bash (from flex template)
+
+Below is an example of launching a dataflow pipeline manually, based on the built template
+
+**Note:** In the data platform demo, the launch of this Dataflow pipeline is handled by the airflow operator (DataflowStartFlexTemplateOperator).
+
+```
+#!/bin/bash
+
+PROJECT_ID=[LOAD-PROJECT]
+REGION=europe-west1
+SERVICE_ACCOUNT=[LOAD-DF-SA]0@ff-lod.iam.gserviceaccount.com
+
+PIPELINE_STAGIN_PATH="gs://[LOAD-STAGING-BUCKET]/build"
+CSV_FILE=gs://[DROP-ZONE-BUCKET]/customers.csv
+JSON_SCHEMA=gs://[ORCHESTRATION-BUCKET]/customers_schema.json
+OUTPUT_TABLE=[DESTINATION-PROJ].[DESTINATION-DATASET].customers
+TEMPLATE_PATH=gs://[ORCHESTRATION-DF-GCS]/csv2bq.json
+
+
+gcloud dataflow flex-template run "csv2bq-`date +%Y%m%d-%H%M%S`" \
```

Comment:
I would impersonate the Orc SA:

`--impersonate-service-account=orchestrator@$SERVICE_PROJECT_ID.iam.gserviceaccount.com`

Reply:
done.

Reply:
This gcloud command is still not using the SA. I would explicitly use the --impersonate-service-account flag for the Google Cloud CLI.

Reply:
Done.